File: backend/smart-library-backend/src/main/java/com/library/service/ReservationService.java
package com.library.service;

import com.library.dto.request.ReservationRequest;
import com.library.dto.response.PageResponse;
import com.library.dto.response.ReservationResponse;
import org.springframework.data.domain.Pageable;

public interface ReservationService {
    ReservationResponse makeReservation(ReservationRequest request, String username);
    ReservationResponse cancelReservation(Long id, String username);
    ReservationResponse fulfillReservation(Long id, String adminUsername);
    PageResponse<ReservationResponse> getAllReservations(Pageable pageable);
    PageResponse<ReservationResponse> getMyReservations(String username, Pageable pageable);
    void processExpiredReservations();
}

File: frontend/smart-library-frontend/src/services/reservationService.js

import api from './api';

export const makeReservation = (data) => api.post('/reservations', data);
export const cancelReservation = (id) => api.put(`/reservations/${id}/cancel`);
export const fulfillReservation = (id) => api.put(`/reservations/${id}/fulfill`);
export const getAllReservations = (page = 0, size = 10) =>
  api.get(`/reservations?page=${page}&size=${size}`);
export const getMyReservations = (page = 0, size = 10) =>
  api.get(`/reservations/my?page=${page}&size=${size}`);
